tests/compiler/test_plugin_types.py

```diff
@@ -17,44 +17,71 @@
 """
 
 import collections.abc
-from typing import Any, Mapping, Sequence, Union
+from typing import Annotated, Any, Mapping, Sequence, Union
 
 import pytest
 
 import inmanta.ast.type as inmanta_type
-from inmanta.ast import RuntimeException
+from inmanta import plugin_typing
+from inmanta.ast import Namespace, Range, RuntimeException
 from inmanta.plugins import Null, to_dsl_type
 
+# from inmanta.ast.entity import Entity
+# from inmanta.ast.type import TYPES
+
 
 def test_conversion(caplog):
     """
     Test behaviour of to_dsl_type function.
     """
-    assert inmanta_type.Integer() == to_dsl_type(int)
-    assert inmanta_type.Float() == to_dsl_type(float)
-    assert inmanta_type.NullableType(inmanta_type.Float()) == to_dsl_type(float | None)
-    assert inmanta_type.List() == to_dsl_type(list)
-    assert inmanta_type.TypedList(inmanta_type.String()) == to_dsl_type(list[str])
-    assert inmanta_type.TypedList(inmanta_type.String()) == to_dsl_type(Sequence[str])
-    assert inmanta_type.List() == to_dsl_type(Sequence)
-    assert inmanta_type.List() == to_dsl_type(collections.abc.Sequence)
-    assert inmanta_type.TypedList(inmanta_type.String()) == to_dsl_type(collections.abc.Sequence[str])
-    assert inmanta_type.TypedDict(inmanta_type.Type()) == to_dsl_type(dict)
-    assert inmanta_type.TypedDict(inmanta_type.Type()) == to_dsl_type(Mapping)
-    assert inmanta_type.TypedDict(inmanta_type.String()) == to_dsl_type(dict[str, str])
-    assert inmanta_type.TypedDict(inmanta_type.String()) == to_dsl_type(Mapping[str, str])
-
-    assert inmanta_type.TypedDict(inmanta_type.String()) == to_dsl_type(collections.abc.Mapping[str, str])
-
-    assert Null() == to_dsl_type(Union[None])
-
-    assert isinstance(to_dsl_type(Any), inmanta_type.Type)
+    namespace = Namespace("dummy-namespace")
+    # namespace.set_primitives(inmanta_type.TYPES)
+    # FIXME: Not working    because of std = self.get_ns_from_string("std") std is None
+    namespace.primitives = inmanta_type.TYPES
+
+    location: Range = Range("test", 1, 1, 2, 1)
+
+    assert inmanta_type.String() == to_dsl_type(plugin_typing.string, location, namespace)
+
+    assert inmanta_type.NullableType(inmanta_type.Integer()) == to_dsl_type(
+        Annotated[int | None, "something"], location, namespace
+    )
+
+    assert inmanta_type.TypedDict(inmanta_type.Type()) == to_dsl_type(
+        Annotated[dict[str, int], plugin_typing.InmantaType("dict")], location, namespace
+    )
+
+    # FIXME: Not working
+    # entity: Entity = Entity("my-entity", namespace)
+    # namespace.define_type("my-entity", entity)
+    # assert inmanta_type.TypedDict(inmanta_type.Type()) == to_dsl_type(
+    #     Annotated[Union[int, str] | None, plugin_typing.InmantaType("my-entity")], location, namespace
+    # )
```

Comment:
It is a bit awkward setting up the namespace in order to do this. Do you have any tips @sanderr ?